tour_model_eval/trips_in_bins_and_clusters.ipynb

```diff
@@ -0,0 +1,288 @@
+   "execution_count": null,
+   "id": "latest-cooling",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "bin_start_lat = []\n",
+    "bin_start_log = []\n",
+    "\n",
+    "# Flip indices because points are in geojson (i.e. lon, lat)\n",
+    "for trip in bin_trips:\n",
+    "    trip_start_lat = trip.data.start_loc[\"coordinates\"][1]\n",
+    "    trip_start_log = trip.data.start_loc[\"coordinates\"][0]\n",
+    "    bin_start_lat.append(trip_start_lat)\n",
+    "    bin_start_log.append(trip_start_log)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "international-unemployment",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bin_end_lat = []\n",
+    "bin_end_log = []\n",
+    "for trip in bin_trips:\n",
+    "    trip_end_lat = trip.data.end_loc[\"coordinates\"][1]\n",
+    "    trip_end_log = trip.data.end_loc[\"coordinates\"][0]\n",
+    "    bin_end_lat.append(trip_end_lat)\n",
+    "    bin_end_log.append(trip_end_log)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "systematic-huntington",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# bin_trips"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "written-block",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logging.debug('For the first trip, the start point is %s' % [bin_start_lat[0], bin_start_log[0]])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "atmospheric-giant",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "logging.debug('For the first trip, the end point is %s' % [bin_end_lat[0],bin_end_log[0]])\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "oriental-notice",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Plot all the bin trips on the map, later we should use different color for different bins, but not sure whether it works.\n",
+    "m = folium.Map(location=[bin_start_lat[0], bin_start_log[0]],zoom_start=12, max_zoom= 30, control_scale=True)\n",
+    "for i in range (len(bin_trips)):\n",
+    "    layer = folium.PolyLine([[bin_start_lat[i],bin_start_log[i]],[bin_end_lat[i],bin_end_log[i]]],weight=2, color='orange')\n",
+    "    layer.add_to(m)\n",
+    "m"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "sonic-essay",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "########clusters#####"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "lightweight-plaintiff",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# cluster the data using k-means\n",
+    "# def cluster(data, bins)\n",
+    "# return feat.clusters, feat.labels, feat.data\n",
+    "clusters,labels,feat_data = pipeline.cluster(bin_trips,bins)\n",
```

Comment:
The error is fairly simple. The original pipeline calls `n, labels, data = cluster(data, len(bins))`.
You are calling `pipeline.cluster(bin_trips,bins)`.

So the original pipeline expects a number as the second argument, which is why it can multiply it by `1.5`
You are passing in a list, which is why you can't multiply it by a float.

Reply:
if you had added some additional logging to `cluster(` you would have been able to discover this quickly!

Reply:
Can you show an example of logging to figure out the what exactly the argument is?
`def cluster(data, bins)`is in the file right after she return bins in the previous function, that's why I pass bins

Reply:
can you add a link to where `def cluster(data, bins)` is?
Here's what I see
https://github.com/e-mission/e-mission-server/blob/master/emission/analysis/modelling/tour_model/cluster_pipeline.py#L87
which has `n, labels, data = cluster(data, len(bins))`

Reply:
Oh, I didn't see the main when I call this function. 
[https://github.com/e-mission/e-mission-server/blob/master/emission/analysis/modelling/tour_model/cluster_pipeline.py#L61](url)

Reply:
@corinne-hcr here is how you would clarify that function + add additional logs
https://github.com/e-mission/e-mission-server/pull/792